### ISSUE TITLE
`editor.experimentalEditContextEnabled` disrupts extensions like vscodevim and vscode-neovim

### ISSUE THREAD

**Issue by wenfangdu** [microsoft/vscode#228223]
This new option disrupts extensions like vscodevim and vscode-neovim. These extensions register commands such as `type` and `replacePreviousChar`.

_Originally posted by @xiyaowong in https://github.com/microsoft/vscode/issues/183814#issuecomment-2343097924_

**Comment by aiday-mar** [comment]
Hi thank you for filing this issue. I am not a user of vscode vim or vscode neo vim. Could you please add exact detailed steps for how to reproduce this issue?

**Comment by wenfangdu** [comment]
@aiday-mar Then it's a great time to learn Vim ;). Here's a Vim cheat sheet, https://vim.rtorr.com/, you can compare the commands with `editor.experimentalEditContextEnabled` on/off, e.g. `gg` is supposed to go to the top of the editor, `G` is supposed to go to the bottom of the editor, with `editor.experimentalEditContextEnabled` on, `gg`/`G` don't act as they should. I'm using [vscode-neovim](https://github.com/vscode-neovim/vscode-neovim), you can start with that too.

**Comment by aiday-mar** [comment]
Hi thanks a lot for the help. I was able to reproduce this issue and after some investigation, I have found that we were not sending the correct payloads in the input events to the commands. The following PR fixes this: https://github.com/microsoft/vscode/pull/229129. 

Please let me know if you continue seeing this issue once the PR is merged, with the VS Code Neovim or Vim commands.

**Comment by aiday-mar** [comment]
After some more investigation, I have made the following PR https://github.com/microsoft/vscode/pull/229166 which I believe now should generally fix all potential issues with extension custom registered commands. One of the issues that the PR fixes is that we were previously setting the cursor states before executing the command, which we should not do, as extensions such as Neovim set the cursor states explicitly. 

When next Insiders comes out, please let me know if you continue seeing issues with the extension with the edit context setting enabled.

**Comment by wenfangdu** [comment]
@aiday-mar Well done, it's much better now, I'll leave this open until I see no issues when using `experimentalEditContextEnabled` with the stable vscode build.

**Comment by wenfangdu** [comment]
@aiday-mar Please see edit.

**Comment by aiday-mar** [comment]
Hi @wenfangdu thanks for the ping. The experimental edit context should be made available in the stable build some time next week when the next stable build comes out.

**Comment by wenfangdu** [comment]
@aiday-mar Thanks for the reminder, I'll use it for some time and get back to you.

**Comment by wenfangdu** [comment]
@aiday-mar As I was using vscode-neovim, I did notice one important issue.

OS: macOS 15
VSCode: 1.94.2
IME: Sogou Pinyin

Repro Steps:
1. Turn on `editor.experimentalEditContextEnabled`
2. Quit vscode and reopen
3. Use IME to input
4. The IME input box misaligns
5. Focus the search panel and then focus back to the editor panel
6. The IME input box aligns correctly

https://github.com/user-attachments/assets/534bddf5-98f4-4c37-8c02-eef7406ddc63

**Comment by wenfangdu** [comment]
@aiday-mar Aiday, how's the progress?

**Comment by aiday-mar** [comment]
Hi @wenfangdu thanks for the ping. This iteration, our efforts as a team have been directed towards more AI applications. I'll look at this when there is time.

**Comment by aiday-mar** [comment]
Hi @wenfangdu I think what you are seeing could maybe be a chromium issue that should be fixed in earlier versions. Could you verify if this bug (or similar behavior) happens on [insiders-vscode.dev](insiders-vscode.dev) on your up-to-date chrome browser?

**Comment by wenfangdu** [comment]
@aiday-mar You mean https://insiders.vscode.dev/ right? [insiders-vscode.dev](https://github.com/microsoft/vscode/issues/insiders-vscode.dev) is not valid. I tried it in the browser and didn't reproduce, but not sure if it applies for the vscode local app. Please merge this fix in the insider build, and I'll test this out.

**Comment by aiday-mar** [comment]
Hi yes I meant the website. Thanks for letting us know.

**Comment by aiday-mar** [comment]
Hi it would seem that the original issue here has been fixed. I will therefore close the issue. Feel free to open a new issue if you see other bugs.
